pmd-java/src/main/java/net/sourceforge/pmd/lang/java/oom/ParameterizedMetricKey.java

```diff
@@ -16,7 +16,7 @@
  *
  * @author Clément Fournier
  */
-public final class ParameterizedMetricKey {
+/* default */ final class ParameterizedMetricKey {
```

Comment:
Nope, it's an internal structure (not part of the API) to represent a metric key coupled with a metric version. It's used to index memoization maps, so that several versions of the same metric can be cached. And because it doesn't care about where its metrickey and version comes from, it allows metrics defined outside the framework (custom ones) to be cached too